UnitTests/GitCommands.Tests/Git/ObjectIdTests.cs

```diff
@@ -306,5 +306,14 @@ public void Equals_using_operator()
             Assert.IsFalse(ObjectId.Parse(objectIdString) == ObjectId.Random());
             Assert.IsTrue(ObjectId.Parse(objectIdString) != ObjectId.Random());
         }
+
+        [Test]
+        public void Equals_against_hash_string()
+        {
+            string objectIdString = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+            Assert.IsTrue(ObjectId.Parse(objectIdString).Equals(objectIdString));
+            Assert.IsFalse(ObjectId.Parse(objectIdString).Equals("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab"));
+            Assert.IsFalse(ObjectId.Parse(objectIdString).Equals("baaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"));
+        }
```

Comment:
I think we need more tests as the cyclomatic complexity of the `Equals` is greater than one (I think, though I'm probably undercounted, it's above 8).